test/boost/expr_test.cc

```diff
@@ -1545,3 +1545,21 @@ BOOST_AUTO_TEST_CASE(prepare_null_no_type_fails) {
     BOOST_REQUIRE_THROW(prepare_expression(null_expr, db, "test_ks", table_schema.get(), nullptr),
                         exceptions::invalid_request_exception);
 }
+
+BOOST_AUTO_TEST_CASE(prepare_bind_variable) {
+    schema_ptr table_schema = make_simple_test_schema();
+    auto [db, db_data] = make_data_dictionary_database(table_schema);
+
+    expression bind_var = bind_variable{.bind_index = 1, .receiver = nullptr};
+
+    ::lw_shared_ptr<column_specification> receiver = make_receiver(int32_type);
+
+    expression prepared = prepare_expression(bind_var, db, "test_ks", table_schema.get(), receiver);
```

Comment:
Should also test the second case (no receiver -> failed type inference)

Reply:
Added, although it turned out that we don't check if the receiver for bind_variable exists x.x
Fixed the bug in the previous commit.